### PR TITLE
Printf migration name and date

### PR DESCRIPTION
**Pull request by lucasmrod**

### CODE REVIEW

File: migration.go [hunk 1]
```diff
@@ -64,13 +65,34 @@ func (c *Client) runMigration(db *sql.DB, m *Migration, direction bool) error {
...
+var uppercaseReplace = regexp.MustCompile("([a-z])([A-Z])")
+
+func parseNameAndDate(source string) (name string, date string) {
+	parts := strings.SplitN(strings.TrimSuffix(filepath.Base(source), ".go"), "_", 2)
+	// Stripping seconds [:8] because Fleet developers add seconds when re-arranging new migrations
+	// e.g.: 2022/10/10 15:43:46 fail to parse time: parsing time "20201021104586": second out of range
+	datePart := parts[0][:8]
+	mt, err := time.Parse("20060102", datePart)
+	if err != nil {
+		log.Fatalf("fail to parse time: %s", err)
```
Comment (chiiph) [inline migration.go:88]: Do we want to fail like this here? Could we just print something different? I guess this would fail on a test early enough.

**Review by chiiph — APPROVED** [review]

File: migration.go (hunk 1 above)
Comment (lucasmrod) [reply migration.go:88]: Correct. Plus all this names are generated in an automated way (via `make migration name=FooBar`).

File: migration.go [hunk 2]
```diff
@@ -64,13 +65,34 @@ func (c *Client) runMigration(db *sql.DB, m *Migration, direction bool) error {
...
+	}
+	name = string(uppercaseReplace.ReplaceAll([]byte(parts[1]), []byte("$1 $2"))) // add spaces in the filename
```
Comment (mna) [inline migration.go]: Nit: `ReplaceAllString` so no need to convert back to `[]byte`.

**Review by mna — APPROVED** [review]
:+1: , just left a comment for a small improvement.

**Review by mna — APPROVED** [review]

**Pull request merged**